lib/daru/view/adapters/googlecharts/display.rb

```diff
@@ -18,23 +19,59 @@ module Display
     # @return [String] The ID of the DIV element that the Google Chart or
     #   Google DataTable should be rendered in
     attr_accessor :html_id
+
     def show_script(dom=SecureRandom.uuid, options={})
       script_tag = options.fetch(:script_tag) { true }
       if script_tag
-        # if it is data table
-        if is_a?(GoogleVisualr::DataTable)
-          to_js_full_script(dom)
-        else
-          to_js(dom)
-        end
+        show_script_with_script_tag(dom)
+      elsif data.is_a?(String)
+        get_html_spreadsheet(data, dom)
+      else
+        get_html(dom)
+      end
+    end
+
+    # @param dom [String] The ID of the DIV element that the Google
+    #   Chart should be rendered in
+    # @return [String] js code to render the chart with script tag
+    def show_script_with_script_tag(dom=SecureRandom.uuid)
+      # if it is data table and importing data from spreadsheet
+      if is_a?(GoogleVisualr::DataTable) && data.is_a?(String)
+        to_js_full_script_spreadsheet(data, dom)
+      elsif is_a?(GoogleVisualr::DataTable)
+        to_js_full_script(dom)
+      # Importing data from spreadsheet
+      elsif data.is_a?(String)
+        to_js_spreadsheet(data, dom)
       else
-        html = ''
-        html << load_js(dom)
-        html << draw_js(dom)
-        html
+        to_js(dom)
       end
     end
 
+    # @param dom [String] The ID of the DIV element that the Google
+    #   Chart should be rendered in
+    # @return [String] js code to render the chart
+    def get_html(dom)
```

Comment:
I think, it is same as `to_js` method.

Reply:
Actually, I was going to ask you about this. I wanted to know why are we generating the googlecharts' code with script tag (by calling `to_js` method) and without script tag (by calling `load_js` and `draw_js`) separately in display.rb. We have already defined script tags in the template so we'll definitely be generating the script without the script tags (using load_js and draw_js)?
`to_js` method is never actually called.